src/components/UserSelector.tsx

```diff
@@ -9,8 +27,14 @@ export const UserSelector: React.FC = () => {
           className="button"
           aria-haspopup="true"
           aria-controls="dropdown-menu"
+          onClick={() => setShowUserList(!showUserList)}
+          onBlur={() => setTimeout(() => setShowUserList(!showUserList), 200)}
```

Comment:
Using `setTimeout` with `onBlur` to toggle the dropdown visibility can lead to unexpected behavior, especially if the user interacts quickly. Consider using a more robust approach to manage focus and dropdown visibility.